crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -268,7 +268,7 @@ async fn payment_response_update_tracker<F: Clone, T>(
         Err(err) => (
             Some(storage::PaymentAttemptUpdate::ErrorUpdate {
                 connector: Some(router_data.connector.clone()),
-                status: storage::enums::AttemptStatus::Failure,
+                status: router_data.status.foreign_into(),
```

Comment:
This will use the same status as provided by the connector processing step for updating the status. referencing this issue for more information #353